### ISSUE TITLE
Incorrect softmax use

### ISSUE THREAD

**Issue by wcarvalho** [shaohua0116/Relation-Network-Tensorflow#5]
You aplpy tf.nn.softmax inside your model and then apply tf.nn.softmax_cross_entropy_with_logits to that softmax:
```
self.all_preds = tf.nn.softmax(logits)
loss = tf.nn.softmax_cross_entropy_with_logits(logits=logits, labels=labels)
```
however, the [tensorflow documentation](https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits) clearly states

"WARNING: This op expects unscaled logits, since it performs a softmax on logits internally for efficiency. Do not call this op with the output of softmax, as it will produce incorrect results."

This seems like a bug?

**Comment by shaohua0116** [comment]
The variable `self.all_preds` is only used for the output. Only logits are used to calculate the loss.

**Comment by wcarvalho** [comment]
Thanks, I wasn't paying attention! 

Cheers :)
